trio/_tests/type_tests/path.py

```diff
@@ -0,0 +1,118 @@
+"""Path wrapping is quite complex, ensure all methods are understood as wrapped correctly."""
```

Comment:
I wish there was a nicer way to check we conform to typeshed's type hints and I think there might be other places where that would benefit us. But I'm not sure really how and it's probably worth punting that to another PR if we ever figure it out.

Reply:
It's tricky, since we're not actually entirely conforming - we switch to async of course, and automatically replace `pathlib.Path` → `trio.Path`. Perhaps the best way might be to do things like code generation to create the tests?

Reply:
Let's just defer on this. Let's merge this PR!